### PR TITLE
register SW when using next@13 app directory

### PR DESCRIPTION
**Pull request by floatingdino**
When using the new `app` directory in Next 13 a new main JS entrypoint is introduced, causing the PWA not to be installed.
This PR also adds the register code to the Next 13 app entry point.

### CODE REVIEW

**Review by timbset — APPROVED** [review]

**Comment by faran4engg** [comment]
Can we please merge this !

**Comment by mavic111** [comment]
It is work, please merge this.

**Comment by zamarawka** [comment]
@shadowwalker is it possible to get it merged?

**Review by HenriqueSchroeder — APPROVED** [review]

**Comment by Carlosarturo28** [comment]
Can this be merged now?

**Comment by Bsignx** [comment]
Can this be merged now?

**Comment by Eliav2** [comment]
any update on this one? the latest version of next-pwa (5.6.0) does not work with next 13.3.0... the service worker is not installed properly.

**Comment by floatingdino** [comment]
Waiting on @shadowwalker to merge this one.

**Comment by alirezavalizade** [comment]
I think @shadowwalker is not around

**Comment by ImBIOS** [comment]
@alirezavalizade @floatingdino @Eliav2 @Bsignx @Carlosarturo28 @zamarawka @mavic111 @faran4engg @paales It seems this repo is dead and the maintainer cannot be contacted for so long now. I've made an immediate response by resurrecting this repo [ImBIOS/next-pwa](https://github.com/ImBIOS/next-pwa).

**Comment by floatingdino** [comment]
@ImBIOS Your fork doesn't seem to address as many of the issues with next@13 app directory as this one: https://github.com/DuCanhGH/next-pwa

**Comment by Haseebshah936** [comment]
Hi, I am facing this issue
No matching service worker detected. You may need to reload the page, or check that the scope of the service worker for the current page encloses the scope and start URL from the manifest.

Also the site is installable on localhost but not install able once deployed on vercel. Does this pull request fix the issue. How can I get benefit from this is there a forked or a separate package that can be used. Or if any one can point me to the direction that how can I make it a PWA by my own as we do in react by registering service worker. Is there any docs or article related to that. Let me know thanks. @floatingdino

And also I have already tried this @imbios/next-pwa package of @ImBIOS and it gives the same error.

**Pull request closed without merging**
